File: src/main/java/com/iivanovs/bookshopca/Interface/CommentService.java
package com.iivanovs.bookshopca.Interface;

import com.iivanovs.bookshopca.entity.Book;
import com.iivanovs.bookshopca.entity.Comment;

import java.util.List;
import java.util.Optional;

public interface CommentService {
    public Book create(long user_id, long book_id, Comment comment);
}

File: src/main/resources/templates/js/_services/comment.service.js
import {authHeader} from '../_helpers';
import axios from 'axios';
import {linkConstants} from '../_constants';

export const commentService = {
    comment
};

function comment(user, book, comment) {
    const requestOptions = {
        method: 'POST',
        url: linkConstants.URL + 'comment?user_id=' + user.id + '&book_id=' + book.id,
        headers: {'Content-Type': 'application/json', ...authHeader()},
        allowedHeaders: ['Accept-Version', 'Authorization', 'Credentials', 'Content-Type'],
        data: comment
    };

    return axios(
        requestOptions
    ).then(response => {
        if (!response.data) {
            return Promise.reject("Can't create comment");
        }
        return response.data;
    }).catch(error => {
        return Promise.reject("Can't create comment");
    });
}